File: androidunsplash/src/main/java/com/kc/unsplash/api/endpoints/AuthEndpointInterface.java
package com.kc.unsplash.api.endpoints;

import com.kc.unsplash.models.Stats;
import com.kc.unsplash.models.Token;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;
import retrofit2.http.Url;

public interface AuthEndpointInterface {

    @POST
    Call<Token> getToken(@Url String url,
                         @Query("client_id") String clientID,
                         @Query("client_secret") String clientSecret,
                         @Query("redirect_uri") String redirectURI,
                         @Query("code") String code,
                         @Query("grant_type") String grantType);
}

File: androidunsplash/src/main/java/com/keenencharles/unsplash/api/endpoints/AuthEndpointInterface.kt
package com.keenencharles.unsplash.api.endpoints

import com.keenencharles.unsplash.models.Token
import retrofit2.Call
import retrofit2.http.POST
import retrofit2.http.Query
import retrofit2.http.Url

interface AuthEndpointInterface {

    @POST
    fun getToken(@Url url: String,
                 @Query("client_id") clientID: String,
                 @Query("client_secret") clientSecret: String,
                 @Query("redirect_uri") redirectURI: String,
                 @Query("code") code: String,
                 @Query("grant_type") grantType: String): Call<Token>
}
